Base content

tswicegood opened the pull request:
Creating a base content model that can be used for cross-model querying.

tswicegood committed:
Add django-taggit and django-model-utils to the requirements

tswicegood committed:
Handle the case where there is no model instance

This was uncovered while integrating with [django-model-utils][].  Any
time a field is accessed directly on a model class `instance` is passed
in as None.  Looking at the Django implemenation, they return the
descriptor in such a case, so I'm doing that here as well

[django-model-utils]: http://pypi.python.org/pypi/django-model-utils

tswicegood committed:
Pull this code in so it can be used to test abstracts

This needs to be moved into armstrong.dev or into an armstrong.test
package so it's not duplicated across all of the apps that need it.

tswicegood committed:
Add in an abstract `ContentBase`

This starts down the path of having a single object to query across.  It
addresses the main issue with having a one-to-one resulting in a bunch
of queries by using model_utils.

tswicegood commented on armstrong/core/arm_content/models.py:
```diff
@@ -1,4 +1,19 @@
+
+class ContentBase(mixins.AuthorsMixin, mixins.PublicationMixin, models.Model):
+    title = models.CharField(max_length=255)
+    summary = models.TextField()
```
Not sure how I feel about `title`/`summary` always being present.

tswicegood commented on armstrong/core/arm_content/models.py:
```diff
@@ -1,4 +1,19 @@
+    title = models.CharField(max_length=255)
+    summary = models.TextField()
+
+    tags = TaggableManager()
```
This is tentative.  From a DRY standpoint, it'd be nice to have it here.

nosamanuel replied:
Maybe it would be worth having a setting for the default publication mixin class?

tswicegood replied:
Possibly...  At first glance I'm ok with having the defaults be the defaults and if you want more you can change them yourself.  For all intents and purposes, `PublicationMixin` is what you should be using and it should have `sites`.  If you deviate from that, you probably want more control anyhow.

tswicegood committed:
First pass at documenting this class

tswicegood committed:
fix link

tswicegood committed:
Add sites to base PublicationMixin and add alternate simplified version

tswicegood committed:
not sure how this wasn't already there

tswicegood committed:
these have been botched for awhile... :-/

tswicegood committed:
add django-reversion to requirements

tswicegood committed:
update the requirements inside setup.py

tswicegood committed:
Add primary_section and sections to the ContentBase model

tswicegood committed:
Add a new admin.fieldsets module for fieldset helpers

Most of the admin interfaces for Armstrong are going to need these added
as elements in the `fieldsets` tuple.

tswicegood committed:
Switch to constants instead of functions

tswicegood committed:
Merge branch 'master' into base-content

tswicegood committed:
Merge remote-tracking branch 'armstrongcms/master' into base-content

Conflicts:
	requirements/base.txt
	requirements/dev.txt

tswicegood committed:
allow secondary sections to be blank

tswicegood committed:
add a slug field

tswicegood commented:
Going to also add a `slug` field to the base content.  I'm trying to figure out how I want to handle doing a `get_by_full_slug` that pays attention to the `primary_section.full_slug` in addition to the base slug on a content item though.  My first thought was to use `contribute_to_class` to add it to the the manager, but that's kind of a pain.  There's no nice clean way to add methods to the default manager.  Might end up with an extra custom manager that's added to `ContentBase` that will do the querying, but then it becomes a matter of what to name it.

tswicegood committed:
add arm_sections in for testing

tswicegood committed:
Add in support for fetching by slug with new SectionSlugManager
